pkg/controllers/execution/execution_controller.go

```diff
@@ -302,8 +303,9 @@ func (c *Controller) syncToClusters(clusterName string, work *workv1alpha1.Work)
 	syncSucceedNum := 0
 	for _, manifest := range work.Spec.Workload.Manifests {
 		workload := &unstructured.Unstructured{}
-		util.MergeLabel(workload, util.ManagedByKarmadaLabel, util.ManagedByKarmadaLabelValue)
 		err := workload.UnmarshalJSON(manifest.Raw)
+		util.MergeLabel(workload, util.ManagedByKarmadaLabel, util.ManagedByKarmadaLabelValue)
+		util.MergeLabel(workload, workv1alpha2.WorkReferenceUID, string(work.UID))
```

Comment:
Merge label after `unmarshall` changes the behavior, as manifest would not have these labels, does that is expected?

Reply:
For the original code, if manifest hasn't the label `ManagedByKarmadaLabel` in line 305, after line 306 logic, there is no such label in `workload`.

So I think we should add the label after `UnmarshalJSON`.  So, the behavior is expected.

Reply:
For the original code, if manifest hasn't the label ManagedByKarmadaLabel in line 305, after line 306 logic, there is no such label in workload.

So I think we should add the label after UnmarshalJSON. So, the behavior is expected.

Reply:
Got it. Thanks.